scout/adapter/mongo/event.py

```diff
@@ -155,6 +155,19 @@ def events(
 
         return self.event_collection.find(query).sort("created_at", pymongo.DESCENDING)
 
+    def case_events_by_verb(self, institute, case, verb):
+        """Return events with a specific verb for a case of an institute
+        Args:
+          institute (dict): an institute id
+          case (dict): a case id
+          verb (dict): an event action verb, example: "dismiss_variant"
+
+        Returns:
+            pymongo.Cursor: Query results
+        """
+        query = {"institute": institute["_id"], "case": case["_id"], "verb": verb}
```

Comment:
We should include category here for two reasons: 1) it will generally keep the number of hits down - no reason to search different categories for dismiss here at least, 2) there is already an index on events for category-verb, which will buy some speedup at least. Also consider adding a very specific index for this query. It will have quite a use count, and events are collecting quickly - not as fast as variants obviously, but still many per case.

Reply:
Good idea! I've included the category in the query now.

There exist already these 2 indexes for the events collection:

"event": [
        IndexModel([("category", ASCENDING), ("verb", ASCENDING)], name="category_verb"),
        IndexModel([("variant_id", ASCENDING)], name="variant_id"),
    ],

I'm not very skilled with indexes, something like this?
IndexModel([("institute", ASCENDING), ("case", ASCENDING), ("category":ASCENDING), ("verb", ASCENDING)], name="case_verb"),

Reply:
Yes like so! We could also try with just the added category as it will hit one of the indices you listed, and add your full suggestion to that old index-overhaul branch. I'll brush that branch up any week now, since we got a nice new contribution to the cancer variantS view, but I don't think we should merge until around when we can accept some downtime/slowness for rebuilding.

Reply:
Ok, let's do that instead. Merging this one, thanks for the review!